crates/service_utils/src/result.rs

```diff
@@ -50,6 +50,28 @@ impl AppError {
             .insert_header(ContentType::json())
             .json(response)
     }
+    pub fn message(&self) -> String {
+        use diesel::result::DatabaseErrorKind as diesel_error_kind;
+        use diesel::result::Error as diesel_error;
+        match self {
+            AppError::ValidationError(msg)
+            | AppError::BadArgument(msg)
+            | AppError::NotFound(msg) => msg.to_owned(),
+            AppError::UnexpectedError(_) => String::from("Something went wrong"),
```

Comment:
Lets stop doing "Something went wrong". Need a better error message here

Reply:
Also log and throw errors if appropriate

Reply:
@Datron The string returned from message function should be same as what string returned using error macro.
For logging , wherever we pass error in these macro like bad_argument!(err) -> we are logging error there itself. Like you can check new function.
This function just helps in getting the message which Apperror sends , so that we can pass the same message into other error types like here serde deserialisation error